mp4parse/src/fallible.rs

```diff
@@ -0,0 +1,513 @@
+//! The types in this module are thin wrappers around the stdlib types to add
+//! support for fallible allocation. Though the fallible allocator is only
+//! enabled with the mp4parse_fallible feature, the API differences from the
+//! stdlib types ensure that all operations which allocate return a Result.
+//! For the most part, this simply means adding a Result return value to
+//! functions which return nothing or a non-Result value. However, these types
+//! implement some traits whose API cannot communicate failure, but which do
+//! require allocation, so it is important that these wrapper types do not
+//! implement these traits.
+//!
+//! Specifically, do not implement any of the following traits:
+//! - Clone
+//! - Extend
+//! - From
+//! - FromIterator
+//!
+//! This list may not be exhaustive. Exercise caution when implementing
+//! any new traits to ensure they won't potentially allocate in a way that
+//! can't return a Result to indicate allocation failure.
+
+#[cfg(feature = "mp4parse_fallible")]
+extern "C" {
+    fn malloc(bytes: usize) -> *mut u8;
+    fn realloc(ptr: *mut u8, bytes: usize) -> *mut u8;
+}
+
+#[cfg(feature = "mp4parse_fallible")]
```

Comment:
Can we drop this? It looks like we unconditionally need PartialEq? And will we always get it regardless of our config because of the prelude?